usersession/agent/rest_api.go

```diff
@@ -305,3 +310,41 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(nil)
 }
+
+func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
+	if ok, resp := validateJSONRequest(r); !ok {
+		return resp
+	}
+
+	decoder := json.NewDecoder(r.Body)
+
+	type finishRefreshNotificationInfo struct {
```

Comment:
Why not use the one defined in client (or vice-versa)? Sorry if that's a dumb question, at first assumed that it would introduce a circular dependency but that doesn't seem to be the case.

Reply:
That seems like a good idea. We do similar with system APIs in `./daemon` importing `./client` to access structs used to transfer data.

Reply:
Makes sense, I was following pendingSnapRefreshInfo case above (which could benefit from similar simplification, but not doing this in this PR). Done for `finishRefreshNotificationInfo`.